modules/apps/wfrp-browser.js

```diff
@@ -178,14 +178,30 @@ export default class BrowserWfrp4e extends Application {
     let packCount = game.packs.size;
     let packCounter = 0;
 
+    BrowserWfrp4e.prototype.documentCache = BrowserWfrp4e.prototype.documentCache || {};
+  
+    async function cacheDocuments(pack, documents) {
+      BrowserWfrp4e.prototype.documentCache[pack.collection] = documents;
+    }
+    
+  
+    async function getCachedDocuments(pack) {
+      if (BrowserWfrp4e.prototype.documentCache.hasOwnProperty(pack.collection)) {
+        return BrowserWfrp4e.prototype.documentCache[pack.collection];
```

Comment:
Using prototype seems a bit awkward, I think something like `game.wfrp4e.documentCache` seems clearer?

Reply:
This can be done, I'll update the PR to include this change.

Reply:
I've gone ahead and added the suggested location for caching. The only current delay that still persists is the delay from rendering the browser, the PR doesn't address that currently.